### ISSUE TITLE
Does this plugin work only with JavaScript and not config.xml for phonegap build?

### ISSUE THREAD

**Issue by infofinity** [phonegap-build/StatusBarPlugin#17]
Hi there, thanks very much for this plugin. We are using phonegap build with cordova 3.3. We added this to config.xml

```
<gap:plugin name="com.phonegap.plugin.statusbar" />
```

Next, if we add this to our HTML (which by the way isn't mentioned in the README):

```
<script type="text/javascript" src="statusbar.js"></script>
```

And if we add this to the deviceready callback:

```
StatusBar.overlaysWebView(false);
StatusBar.backgroundColorByHexString("#321e15");
StatusBar.styleLightContent();
```

Then things work. However, the README suggests that this can be done just through config.xml. Is that correct? We tried this:

```
<preference name="StatusBarOverlaysWebView" value="false" />
<preference name="StatusBarBackgroundColor" value="#321e15" />
```

But it didn't work.

**Comment by bau720123** [comment]
@infofinity 
+1
the "StatusBarOverlaysWebView" and "StatusBarBackgroundColor" seem not suitable for the phonegap build 3.3's config.xml (for now)
maybe we could find other solution

**Comment by wildabeast** [comment]
I don't see anything in the readme about support for configuring the status bar plugin via config.xml preference tags? Don't think this is supported.

You don't need to include the `<script>` tag as the it is loaded by the cordova script loader, defined using [the `js-module` tag in the plugin.xml](https://github.com/phonegap-build/StatusBarPlugin/blob/master/plugin.xml#L20).

Closing.

**Comment by bau720123** [comment]
to @wildabeast 
@infofinity seem See this URL (I know it is not suitable by PGB 3.3's config.xml)
https://github.com/phonegap-build/StatusBarPlugin/blob/master/plugin.xml

you can find the keywords "StatusBarOverlaysWebView" and "StatusBarBackgroundColor"

**Comment by orenagiv** [comment]
Why is this issue closed?
Is there a way to configure the initial status-bar preferences through the config.xml? (otherwise the App always starts with the default white status-bar, and only once the JS is triggered - the custom preferences become affective).

**Comment by wildabeast** [comment]
I closed the issue because I interpreted the question as whether the StatusBarPlugin could be configured using a preference tag in PhoneGap Build's config.xml, and the answer is no.

The config.xml being mentioned in in the plugin.xml file is the platform-specific iOS config.xml (confusing I know), not the one used to configure your app on PhoneGap Build.

Is there a way to configure the initial status-bar preferences through the config.xml:
Note that you can [modify the iOS Info.plist via the config-file element](http://docs.build.phonegap.com/en_US/3.1.0/configuring_config_file_element.md.html#Config%20File%20Elements). You could potentially use this feature to change the way the StatusBar appears on startup, for example [hiding the Status Bar on startup](https://github.com/apache/cordova-plugins/tree/master/statusbar#hiding-at-startup). Something like this:

```
<gap:config-file platform="ios" parent="UIStatusBarHidden">
    <true/>
</gap:config-file>

<gap:config-file platform="ios" parent="UIViewControllerBasedStatusBarAppearance">
    < false/>
</gap:config-file>
```

**Comment by orenagiv** [comment]
Thanks!
Tried that - and that works, but the webview size is still extending beyond the screen size if the StatusBar plugin is used (but that's another issue).

Anyway, what I'm trying to achieve is a black status-bar, so I also tried this:

```
<gap:config-file platform="ios" parent="UIStatusBarStyle" overwrite="true">
    <array>
        <string>UIStatusBarStyleBlackOpaque</string>
    </array>
</gap:config-file>
```

But there was no change to the color of the StatusBar.
Any idea why?

**Comment by orenagiv** [comment]
Found the issue - my background color was white - so the StatusBar did change the text color, but it appeared white on white. 

I use the background color plugin of PGB to set the color.

Instead, I now use what you described above - I start the App with statusBar hidden - and the show it when the Cordova is ready.
